test/integration/logs.test.js

```diff
@@ -0,0 +1,221 @@
+          }
+          expect(res).to.have.status(200);
+          expect(res.body.message).to.equal("Logs fetched successfully");
+          expect(res.body.logs).to.have.lengthOf(3);
+          const log = res.body.logs[0];
+          expect(log).to.have.property("meta");
+          expect(log).to.have.property("body");
+          expect(log.type).to.equal("REQUEST_CREATED");
+          return done();
+        });
+    });
+
+    it("should return 401 if user is not logged in", function (done) {
+      chai
+        .request(app)
+        .get("/logs?dev=true")
+        .end(function (err, res) {
+          if (err) {
+            return done(err);
+          }
+          expect(res).to.have.status(401);
+          return done();
+        });
+    });
+
+    it("should return 401 if user is authorized to access the logs", function (done) {
+      chai
+        .request(app)
+        .get("/logs")
+        .set("cookie", `${cookieName}=${authToken}`)
+        .end(function (err, res) {
+          if (err) {
+            return done(err);
+          }
+          expect(res).to.have.status(401);
+          expect(res.body.error).to.equal("Unauthorized");
+          expect(res.body.message).to.equal(`You are not authorized for this action.`);
+          return done();
+        });
+    });
+
+    it("should return 400, if user has access and dev flag is not", function (done) {
+      chai
+        .request(app)
+        .get("/logs")
+        .set("cookie", `${cookieName}=${superUserToken}`)
+        .end(function (err, res) {
+          if (err) {
+            return done(err);
+          }
+          expect(res).to.have.status(400);
+          expect(res.body.error).to.equal("Bad Request");
+          expect(res.body.message).to.equal(`Please use feature flag to make this request!`);
+          return done();
+        });
+    });
+
+    it("should return all Logs", function (done) {
+      chai
+        .request(app)
+        .get("/logs?dev=true")
+        .set("cookie", `${cookieName}=${superUserToken}`)
+        .end(function (err, res) {
+          if (err) {
+            return done(err);
+          }
+          expect(res).to.have.status(200);
+          expect(res.body.message).to.equal("All Logs fetched successfully");
+          expect(res.body.data).to.lengthOf(7);
+          return done();
+        });
+    });
+
+    it("should return all formatted Logs", function (done) {
+      chai
+        .request(app)
+        .get("/logs?dev=true&format=feed")
+        .set("cookie", `${cookieName}=${superUserToken}`)
+        .end(function (err, res) {
+          if (err) {
+            return done(err);
+          }
+          expect(res).to.have.status(200);
+          expect(res.body.message).to.equal("All Logs fetched successfully");
+          expect(res.body.data).to.lengthOf(7);
+          return done();
+        });
+    });
+
+    it("should return logs of type = extensionRequests with status code 200", function (done) {
```

Comment:
- are there any other status codes that can be returned if the type is `extensionRequest` if not can we remove the `with status code 200` text from the test name?

Reply:
done.